src/SimpleGraphs/simplegraph.jl

```diff
@@ -0,0 +1,25 @@
+"""
+$(TYPEDEF)
+
+A type representing an undirected graph with unweighted, non-multiple edges.
+
+# Fields
+
+$(TYPEDFIELDS)
+"""
+mutable struct SimpleGraph{T<:Integer} <: AbstractGraph{T,SimpleEdge{T}}
+    """Number of edges"""
+    ne::Int
+    "Adjacency list such that `L[v]` contains the neighbors of vertex `v`"
+    adjlist::Vector{Vector{T}}
+end
+
+GraphsBase.is_directed(::Type{<:SimpleGraph}) = false
+
+GraphsBase.vertices(g::SimpleGraph{T}) where {T} = one(T):T(length(g.adjlist))
+
+function GraphsBase.out_edges(g::SimpleGraph{T}, u) where {T}
+    return (SimpleEdge{T}(u, v) for v in g.adjlist[u])
+end
+
+GraphsBase.in_edges(g::SimpleGraph, v) = out_edges(g, v)
```

Comment:
What about `reverse.(out_edges(g, v))`?

Something I found slightly annoying in `Graphs.jl` was that `SimpleEdge` was sometimes treated as directed and sometimes not, which leads to weird corner cases when designing code that is generic between undirected and directed graphs. Ideally, I think an undirected graph should act as much like a directed graph that has edges in both directions between vertices that have edges. So from that perspective it may be helpful to have `in_edges` and `out_edges` output edges with reverse directions.

Reply:
https://github.com/JuliaGraphs/GraphsBase.jl/issues/16